File: core/src/main/java/com/lomekwi/cave/task/Task.java
package com.lomekwi.cave.task;

import com.lomekwi.cave.app.App;

public interface Task extends Runnable,AutoCloseable{
    /**
     * [0,1]
     */
    float getProgress();
    default String getName(){
        return toString();
    }
    default TaskPool getPool(){
        return App.taskPool;
    }
}

File: core/src/main/scala/com/lomekwi/cave/task/Task.scala
package com.lomekwi.cave.task

import com.lomekwi.cave.app.App

trait Task extends Runnable with AutoCloseable {
  /**
   * [0,1]
   */
  def getProgress(): Float
  def getName(): String = {
    toString()
  }
  def getPool(): TaskPool = {
    App.taskPool
  }
}
